day2/src/main.rs

```diff
@@ -0,0 +1,97 @@
+use std::cmp::{Ord, Ordering, PartialOrd};
+use std::io::prelude::*;
+use std::fs::File;
+
+struct Dimensions {
+    length: usize,
+    width: usize,
+    height: usize,
+}
+
+impl Dimensions {
+    fn new(line: String) -> Dimensions {
+        let (length, width, height) = Self::parse_line(line);
+        Dimensions {
+            length: length,
+            width: width,
+            height: height,
+        }
+    }
+
+    fn parse_line(line: String) -> (usize, usize, usize) {
+        let split = line.trim().split('x').collect::<Vec<_>>();
+        (split[0].parse().unwrap(),
+         split[1].parse().unwrap(),
+         split[2].parse().unwrap())
+    }
+
+    fn surface_area(&self) -> usize {
+        2 * self.length * self.width +
+            2 * self.width * self.height +
+            2 * self.height * self.length
+    }
+}
+
+impl Ord for Dimensions {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let diff = self.surface_area() - other.surface_area();
+        match diff {
```

Comment:
NVM i am a ginormous dumbass, because I implemented Ord/PartialOrd/Eq/Partial eq so I could use Vec::sort instead of Vec::sort_by. Anyways, still interested in this issue.